examples/simple_url.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	"github.com/Clever/optimus/transformer"
+	"gopkg.in/Clever/optimus.v3"
+	jsonSink "gopkg.in/Clever/optimus.v3/sinks/json"
+	csvSource "gopkg.in/Clever/optimus.v3/sources/csv"
+)
+
+/*
+In this example, we will:
+1) read in data from CSV
+2) filter and transform that data in a simple way
+3) write the data to JSON
+4) output the number of rows we read in and the number of rows after filtering
+
+It's possible make an even simpler example, but this is a little more interesting.
+*/
+
+var (
+	inputFilename  = filepath.Join("examples", "urls.csv")
+	outputFilename = filepath.Join("examples", "urls.json")
+	cleverRegex, _ = regexp.Compile("clever.com.*")
+)
+
+// helper function to keep things clean
+func fatalIfErr(err error) {
+	if err != nil {
+		log.Fatalf("error in URL example! Message is: %s", err)
+	}
+}
+
+// filterItems will only output SSL=true URL=/clever.com.*/ entries
+// if SSL or URL are missing, don't bother erroring and instead drop the row
+func filterItems(row optimus.Row) (bool, error) {
+	url, ok := row["url"]
+	if !ok || !cleverRegex.MatchString(url.(string)) {
```

Comment:
Do you think it would be better to use `url.Parse` and check the host value instead of a regex?